uPortal-security/uPortal-security-core/src/main/java/org/apereo/portal/security/oauth/IdTokenFactory.java

```diff
@@ -283,8 +283,16 @@ public String createUserInfo(String username) {
         if (groupMember != null) {
             Set<IEntityGroup> ancestors = groupMember.getAncestorGroups();
             for (IEntityGroup g : ancestors) {
-                if (groupsWhitelist.contains(g.getName())) {
-                    groups.add(g.getName());
+                // Is an override set of groups provided?
+                if (overrideGroups != null) {
```

Comment:
I think this bit needs to be reworked slightly.  (I'm happy to do it... about to start, in fact.)

I don't think it should be possible to obtain _more_ information (than the items specified in the config) in the OIDC token through a query string parameter.